src/libraries/System.Text.Json/tests/Common/ReferenceHandlerTests/ReferenceHandlerTests.Deserialize.cs

```diff
@@ -1514,12 +1514,12 @@ public async Task JsonObjectNonCollectionTest()
                 ""\u0024test"": ""test""
             }";
 
-            // \u0024.* Valid (i.e: \u0024test)
-            EmployeeExtensionData employee = await Serializer.DeserializeWrapper<EmployeeExtensionData>(json, s_deserializerOptionsPreserve);
-            Assert.Equal("test", ((JsonElement)employee.ExtensionData["$test"]).GetString());
+            // Escaped JSON properties still not valid
```

Comment:
Updates the test to account for property unescaping now taking place.

Reply:
What's the motivation for now unescaping the metadata properties? IIRC, we choose not to, in order to use escaping as a workaround for users that wanted to use ReferenceHandler.Preserve with the reserved properties.

See https://github.com/dotnet/runtime/blob/main/src/libraries/System.Text.Json/docs/ReferenceHandling_spec.md#json-objects-if-not-enumerable-class--struct--dictionary---on-deserialize-and-serialize.

Reply:
Thanks, I had assumed this was intentional but I hadn't seen that doc.

Couple of reasons, since polymorphism metadata property names can be user-defined (and therefore assuming the property name is escaped it would be impossible to round-trip). Secondly, it's inconsistent with how property names are handled elsewhere so that workaround happens to work for reference preservation but it doesn't for regular property binding.

I'd be curious to hear your thoughts on the matter, but I'd be inclined to remove this behavior (and perhaps file a breaking change issue).

Reply:
> it's inconsistent with how property names are handled elsewhere

Yes, it was inconsistent on purpose because metadata properties are not meant to be treated as regular properties by the serializer, the Utf8JsonReader does treat them normally though.

It's probably OK to change but my current understanding is that no user is complaining about this, is that correct?

Reply:
No user-reported issues just yet, but it is for example failing to deserialize the following example correctly:

```C#
Base result = JsonSerializer.Deserialize<Base>("""{"naïve" : "derived" }""");
Console.WriteLine(result is Derived); // False

[JsonPolymorphic(TypeDiscriminatorPropertyName = "naïve")]
[JsonDerivedType(typeof(Derived), "derived")]
public record Base;
public record Derived : Base;
```

I can revert this if you feel that it's out of scope, but this seems to be a bug in my view.

Reply:
No need to revert, it looks like a change for the better, assuming a breaking change is filed. We also have plenty of time to expect customer reports in case someone is affected. Thanks for explaining and sharing that scenario.